### ISSUE TITLE
examples with arrays are not handled correctly

### ISSUE THREAD

**Issue by ksuszka** [Swagger2Markup/swagger2markup#154]
Swagger2Markup version: 1.0.0

Problem description:
When json array is given as an example it is stripped from generated markup document.

Isolated swagger specification which shows the issue:
```json
{
  "info": {
    "title": "foo"
  },
  "swagger": "2.0",
  "paths": {
    "/bar": {
      "patch": {
        "parameters": [
          {
            "name": "body",
            "in": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/SamplePayload"
            }
          }
        ]
      }
    }
  },
  "definitions": {
    "SamplePayload": {
      "type": "object",
      "example": [
        {
          "op": "replace",
          "path": "/s",
          "v": "w"
        }
      ]
    }
  }
}
```

Generated markup:
```asciidoc
= foo


[[_overview]]
== Overview



[[_paths]]
== Paths

[[_bar_patch]]
=== PATCH /bar

==== Parameters

[options="header", cols=".^1,.^3,.^10,.^4,.^2"]
|===
|Type|Name|Description|Schema|Default
|*Body*|*body* +
_required_||<<_samplepayload,SamplePayload>>|
|===


==== Example HTTP request

===== Request body
[source,json]
----
""
----




[[_definitions]]
== Definitions

[[_samplepayload]]
=== SamplePayload
_Type_ : object




```

**Comment by Kabhal** [comment]
Are you sure this is valid to provide an 'array of object' as an example for an 'object' ?
It would make sense only if SamplePayload was of type 'array of object'.

**Comment by Kabhal** [comment]
Just try :
```json
{
  "info": {
    "title": "foo"
  },
  "swagger": "2.0",
  "paths": {
    "/bar": {
      "patch": {
        "parameters": [
          {
            "name": "body",
            "in": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/SamplePayload"
            }
          }
        ]
      }
    }
  },
  "definitions": {
    "SamplePayload": {
      "type": "object",
      "properties": {
        "op": { "type": "string" },
        "path": { "type": "string" },
        "v": { "type": "string" }
      },
      "example": {
          "op": "replace",
          "path": "/s",
          "v": "w"
        }
    }
  }
}
```

**Comment by ksuszka** [comment]
I have endpoint which requires [JSON Patch](http://jsonpatch.com/) array as a payload. It has to be an array not an object as it is specified in RFC.
From my experiments given payload type doesn't matter, but just to prove that it is not the case:
```json
{
  "info": {
    "title": "foo"
  },
  "swagger": "2.0",
  "paths": {
    "/bar": {
      "patch": {
        "parameters": [
          {
            "name": "body",
            "in": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/SamplePayload"
            }
          }
        ]
      }
    }
  },
  "definitions": {
    "SamplePayload": {
      "type": "array",
      "items": {
        "type": "object"
      },
      "example": [
        {
          "op": "replace",
          "path": "/s",
          "v": "w"
        }
      ]
    }
  }
}
```

Generated markup:

```
= foo


[[_overview]]
== Overview



[[_paths]]
== Paths

[[_bar_patch]]
=== PATCH /bar

==== Parameters

[options="header", cols=".^1,.^3,.^10,.^4,.^2"]
|===
|Type|Name|Description|Schema|Default
|*Body*|*body* +
_required_||<<_samplepayload,SamplePayload>>|
|===


==== Example HTTP request

===== Request body
[source,json]
----
""
----




[[_definitions]]
== Definitions

[[_samplepayload]]
=== SamplePayload
_Type_ : < object > array






```

**Comment by Kabhal** [comment]
ok, I'll take a look

**Comment by Kabhal** [comment]
First constatations :
* Swagger parser is buggy as getExample() returns ```""``` when an example is set at array type level 
* if we create an example at items type level when upper type is an array, Swagger parser is buggy as getExample() returns ```null```
=> I don't expect a fast solution :/

**Comment by Kabhal** [comment]
As a workaround, you can use Swagger2Markup extensions to include some extra content with your own example using pure AsciiDoc for this path.

**Comment by ksuszka** [comment]
Ok. Thanks for looking into it.

**Comment by Kabhal** [comment]
blocked by https://github.com/swagger-api/swagger-parser/issues/255

**Comment by Kabhal** [comment]
https://github.com/swagger-api/swagger-parser/issues/255 has been resolved partially in swagger-parser:1.0.21-SNAPSHOT
If you really need it, you can use the unofficial branch feature/swagger-parser-1.0.21 .
The feature branch contains a workaround for partially resolved swagger-parser bug.
